packages/mimir/src/rules/parameter-properties.ts

```diff
@@ -0,0 +1,193 @@
+import * as ts from 'typescript';
+import { excludeDeclarationFiles, ConfigurableRule, Replacement } from '@fimbul/ymir';
+import {
+    getLineBreakStyle,
+    isCallExpression,
+    isExpressionStatement,
+    isClassDeclaration,
+    isClassExpression,
+    isConstructorDeclaration,
+    isParameterProperty,
+    isBinaryExpression,
+    isPropertyAccessExpression,
+    isIdentifier,
+    isPropertyDeclaration,
+    hasSideEffects,
+} from 'tsutils';
+
+export interface Options {
+    mode: 'when-possible' | 'never' | 'consistent';
+}
+
+@excludeDeclarationFiles
+export class Rule extends ConfigurableRule<Options> {
+    protected parseOptions(options: Options | null | undefined): Options {
+        return options || { mode: 'when-possible' };
+    }
+
+    public apply() {
+        const checkClass = (node: ts.Node): void => {
+            if (
+                (isClassDeclaration(node) || isClassExpression(node)) &&
+                node.members.filter(isConstructorDeclaration).length === 1
+            )
+                this.checkConstructorParameters(node.members.find(isConstructorDeclaration)!, node);
+
+            return ts.forEachChild(node, checkClass);
+        };
+        ts.forEachChild(this.context.sourceFile, checkClass);
```

Comment:
I prefer to avoid `forEachChild` where possible. That's why I implemented `this.context.getFlatAst()` and `this.context.getWrappedAst()`.
In this case I would iterate the flattened AST and look for constructor declarations with body. The parent will always be a class declaration or expression.